### ISSUE TITLE
Optional per-subproject `turbo.json` file to specify custom dependencies

### ISSUE THREAD

**Issue by iiian** [vercel/turbo#3370]
### Are there functional work arounds to this problem?

Yes, and frankly I am not sure that this issue goes beyond cosmetics.

### Which project is this feature idea for?

Turborepo

### Describe the feature you'd like to request

**BACKGROUND:**
So, as part of my CI/CD lifecycle, I've got a task where I need to apply db migrations in a number of environments (dev/test/prod) for a number of different customers. I have N `services` in `apps`, each of which has its own set of migration scripts. My CI/CD solution is to collect all of the migrations into a single container, and run this container one-off with a configuration to target (customer, environment) for all services.

**APPROACH THAT LED TO FEATURE REQUEST:**
To build this `migration-service`, it needs to be able to pull in those migration sets just prior to build. This is what `turborepo` excels at, although in this case there are some interesting consequences:

1) Either I have to declare a `turbo.json`-level (thus project-wide) dependency: `turbo run build` depends on `turbo run copy-migration-scripts`. This "feels" wrong, because it's setting up blanket-policy to handle a minority situation. If I keep tacking on little rules like this at the top-level, what could be the long term consequences on performance or correctness of the task dependency graph?

Or

2) in the actual `migration-service`'s `package.json:scripts.build` command, I could do `turbo-in-turbo`, which also seems slightly wrong:

```
  "scripts": {
    "build": "(cd <to-project-root> && turbo run copy-migration-scripts); ...continue-with-rest-of-build...",
  }
```

I'm not in love with either.

### Describe the solution you'd like

To address this, what if it were possible to have downstream `turbo.json` files that extend the dependency graph, at the same level as my migration-service. Then I could say

```jsonc
// apps/.../migration-service/turbo.json
{
  "$schema": "https://turborepo.org/schema.json",
  // perhaps something here declaring which turbo.json we are inheriting from
  "pipeline": {
    "build": {
      // note copy-migration-scripts IS declared at root-level of project bc multiple services must specify it
      "dependsOn": [ "copy-migration-scripts" ]
    }
  }
}
```

### Describe alternatives you've considered

1) Declare in `turbo.json` that, projectwide, super common task `build` is dependent on rare minority job `do-special-thing`, which doesn't feel right, but works.

2) In the subproject with the minority job dependency, do `turbo-in-turbo`:
```
  "scripts": {
    "build": "(cd <to-project-root> && turbo run do-special-thing); ...continue-with-rest-of-build...",
  }
```
I call it `turbo-in-turbo` because it's like `docker-in-docker`: when I `turbo run build --scope=migration-service` I get a fun little line in the build output:

```log
migration-service:build: document-service:copy-migration-scripts:
```

3) I could use `turbo prune` with a filter to pick up all `service`s, and then using a multi-stage build, to grab the migration scripts of each `service`. This works for docker builds, but it doesn't work great for local development, although I question if that's terribly important for this project.

I'm not in love with any of these options, although they all work.

**Comment by mehulkar** [comment]
@iiian you get the award for clairvoyance! We've been working on spec'ing out this feature for a couple months now, and I just started implementation in #2706. WE're expecting this to be part of the 1.8 release!

**Comment by iiian** [comment]
Hey that's awesome! You guys develop features right before I need them, but *I'm* the one who is precognitive eh?

**Comment by mehulkar** [comment]
@iiian check out https://turbo.build/repo/docs/core-concepts/monorepos/configuring-workspaces! We released the feature in 1.8: https://turbo.build/blog/turbo-1-8-0
